parser/lexer.go

```diff
@@ -88,7 +89,9 @@ func (s *Scanner) Errors() (warns []error, errs []error) {
 
 // reset resets the sql string to be scanned.
 func (s *Scanner) reset(sql string) {
-	s.r = reader{s: sql, p: Pos{Line: 1}}
+	s.client = charset.EncodingUTF8Impl
```

Comment:
How about `s.client = charset.FindEncoding(mysql.DefaultCharset)`?

Reply:
I think we can use `charset.EncodingBinaryImpl` instead and remove line 151.